nixos/tests/without-nix.nix

```diff
@@ -0,0 +1,37 @@
+{
+  system ? builtins.currentSystem,
+  pkgsFun ? args: import ../.. ({ inherit system; config = {}; } // args),
+  ...
+} @ args:
+
+let
+  # We could simply use the `nixpkgs.overlays` NixOS option, but this more
+  # throughly excises Nix at eval time.
+  pkgs = pkgsFun {
+    overlays = [
+      (self: super: {
+        nix = throw "don't want to use this";
+      })
+    ];
+  };
+in
+
+with import ../lib/testing-python.nix { inherit system pkgs; };
```

Comment:
I don't think `pkgs` makes it all the way into the NixOS config, similar to https://github.com/NixOS/nixpkgs/issues/55798.

If it does, you could prove this important test assumption by adding another attribute in your overlay and adding an assertion about it in the NixOS config.

Reply:
The `pkgs` inherited here is the only one available in `../lib/testing-python.nix`, so why not?